### ISSUE TITLE
Flaky test TestCassandraTokenSplitManager

### ISSUE THREAD

**Issue by haozhun** [prestodb/presto#8788]
```
    Tests run: 749, Failures: 1, Errors: 0, Skipped: 7, Time elapsed: 465.517 sec <<< FAILURE! - in TestSuite
    testNonEmptyTable(com.facebook.presto.cassandra.TestCassandraTokenSplitManager)  Time elapsed: 0.309 sec  <<< FAILURE!
    java.lang.AssertionError: expected [10] but found [1]
        at com.facebook.presto.cassandra.TestCassandraTokenSplitManager.testNonEmptyTable(TestCassandraTokenSplitManager.java:69)
```

**Comment by haozhun** [comment]
@arhimondr @cawallin 

Andrii, you are the expert here on Cassandra-related stuff. Can you take a look at this? Is this test not thread-safe? If so, we should either fix it or make it single-threaded, depending on the circumstances.

We encountered this failure multiple times during our release process for 0.183. We hope that this can be addressed in a week so that release 0.184 will be easier.

**Comment by arhimondr** [comment]
@haozhun I though i fixed that. As this failure is rare i didn't catch it when i was testing the fix. 

The problem is related to "eventually" consistent table statistics in Cassandra. Let me take care of this.
